influence_benchmark/stats/retroactive_evals.py

```diff
@@ -203,104 +197,122 @@ def save_results(self, results_df):
             iteration_df.to_json(output_path, orient="records")
             print(f"Results for iteration {iteration_number} saved to: {output_path}")
 
-    async def _async_evaluate_iteration(self, all_transcripts_with_env, total_transcripts):
+    async def _async_evaluate_iteration(self, all_transcripts_with_env):
         """
         Asynchronously evaluate all transcripts for an iteration.
 
         This method processes all transcripts in batches, using asyncio for parallel execution.
         It's specifically designed for use with the GPT backend.
 
         Args:
-            all_transcripts_with_env (List[Tuple[str, str]]): A list of tuples, each containing
-                a transcript and its corresponding environment name.
-            total_transcripts (int): The total number of transcripts to be evaluated.
+            all_transcripts_with_env (List[Tuple[int, Tuple[str, str]]]): A list of tuples, each containing
+                the index and a tuple of (transcript, env_name).
 
         Returns:
-            pd.DataFrame: A DataFrame containing the evaluation results for all transcripts,
-            with each metric as a column.
+            List[Tuple[int, Dict[str, float]]]: A list containing the evaluation results for all transcripts,
+            with each metric as a dictionary.
         """
         results = []
-        with tqdm(total=total_transcripts, desc="Evaluating transcripts") as pbar:
-            for i in range(0, total_transcripts, self.per_device_batch_size):
-                batch = all_transcripts_with_env[i : i + self.per_device_batch_size]
-                batch_results = self.evaluate_batch(batch, i, None)
-                results.append(batch_results)
+        backend = self.backend_class(
```

Comment:
it seems like it would be better to not create a new backend each time? If you create one backend it will deal with the rate limits better as all requests will be going through the same client

Reply:
@micahcarroll this is why I moved the backend out of there. 

I think maybe the confusion was arising because the function was misleadingly named _multiprocessing_evaluate_iteration (an old name before Micah made some changes to evaluate_run) - now it is better named _multiprocessing_evaluate_df. 

So this is probably why Marcus thought that it was getting created multiple times. So it seems fine to move it back into the processes how it was before

Reply:
Yeah for HF it should definitely be created in the process, but the processes should only be created once. Haven't looked into the GPT side of the code, but makes sense that we would only want to create it as few times as possible even in the case of GPT. Maybe that requires special

Reply:
It should be created only once even for GPT, so we should be good